modules/hal_nordic/nrfs/backends/nrfs_backend_ipc_service.c

```diff
@@ -219,6 +229,34 @@ int nrfs_backend_wait_for_connection(k_timeout_t timeout)
 	return (events == IPC_INIT_DONE_EVENT ? 0 : (-EAGAIN));
 }
 
+static bool node_uniqueness_validate(sys_slist_t *slist, sys_snode_t *new_node)
+{
+	sys_snode_t *current_node;
+
+	SYS_SLIST_FOR_EACH_NODE(slist, current_node) {
+		if (current_node == new_node) {
+			return false;
+		}
+	}
+
+	return true;
+}
+
+int nrfs_backend_register_bound_cb(struct nrfs_backend_bound_info_cb *cb)
+{
+	if (!cb) {
+		return -EINVAL;
+	}
+
+	if (!node_uniqueness_validate(&nrfs_backend_info_cb_slist, &cb->node)) {
+		return 0;
+	}
```

Comment:
is this really needed? the struct should be treated as opaque, and callback given + assigned in the call. So consumers of the API should never touch the struct